src/lib/Transform.ts

```diff
@@ -229,11 +236,27 @@ export function batch<T>(
 				flushError = e;
 			}
 
-			await writable.end(error, readable.result().catch(noop));
-			throwIfThrowable(earlyFlushError);
-			throwIfThrowable(flushError);
+			const toThrow = earlyFlushError || flushError;
```

Comment:
This is indeed nicer, because it's more clear that we will only throw a single error, instead of the two separate `throwIfThrowable()` lines.

Reply:
I agree! I've now done the same in `forEach()`, the other place where this case shows up.